cirq-core/cirq/devices/superconducting_qubits_noise_properties.py

```diff
@@ -0,0 +1,269 @@
+            targets) to the Pauli error for that operation. Keys in this dict
+            must have defined qubits.
+        validate: If True, performs validation on input arguments. Defaults
+            to True.
+    """
+
+    gate_times_ns: Dict[type, float]
+    t1_ns: Dict['cirq.Qid', float]
+    tphi_ns: Dict['cirq.Qid', float]
+    ro_fidelities: Dict['cirq.Qid', List[float]]
+    gate_pauli_errors: Dict[OpIdentifier, float]
+
+    validate: bool = True
+    _qubits: List['cirq.Qid'] = field(init=False, default_factory=list)
+    _depolarizing_error: Dict[OpIdentifier, float] = field(init=False, default_factory=dict)
+
+    def __post_init__(self):
+        if not self.validate:
+            return
+        t1_qubits = set(self.t1_ns)
+        tphi_qubits = set(self.tphi_ns)
+
+        if t1_qubits != tphi_qubits:
+            raise ValueError('Keys specified for T1 and Tphi are not identical.')
+
+        # validate two qubit gate errors.
+        self._validate_symmetric_errors('gate_pauli_errors')
+
+    def _validate_symmetric_errors(self, field_name: str) -> None:
+        gate_error_dict = getattr(self, field_name)
+        for op_id in gate_error_dict:
+            if len(op_id.qubits) != 2:
+                # single qubit op_ids also present, or generic values are
+                # specified. Skip these cases
+                if len(op_id.qubits) > 2:
+                    raise ValueError(
+                        f'Found gate {op_id.gate_type} with {len(op_id.qubits)} qubits. '
+                        'Symmetric errors can only apply to 2-qubit gates.'
+                    )
+            elif op_id.gate_type not in self.two_qubit_gates():
+                raise ValueError(
+                    f'Found gate {op_id.gate_type} which does not appear in the '
+                    'symmetric or asymmetric gate sets.'
+                )
+            else:
+                # TODO: this assumes op is symmetric.
+                # If asymmetric gates are added, we will need to update it.
+                op_id_swapped = OpIdentifier(op_id.gate_type, *op_id.qubits[::-1])
+                if op_id_swapped not in gate_error_dict:
+                    raise ValueError(
+                        f'Operation {op_id} of field {field_name} has errors '
+                        f'but its symmetric id {op_id_swapped} does not.'
+                    )
+
+    @property
+    def qubits(self) -> List['cirq.Qid']:
+        """Qubits for which we have data"""
+        if not self._qubits:
+            self._qubits = sorted(self.t1_ns)
+        return self._qubits
+
+    @classmethod
+    def single_qubit_gates(cls) -> Set[type]:
+        return SINGLE_QUBIT_GATES
+
+    @classmethod
+    def two_qubit_gates(cls) -> Set[type]:
+        return TWO_QUBIT_GATES
+
+    @classmethod
+    def expected_gates(cls) -> Set[type]:
+        return cls.single_qubit_gates() | cls.two_qubit_gates()
```

Comment:
Gates that this type recognizes ("expects"). We need a method instead of referencing `(SINGLE|TWO)_QUBIT_GATES` directly so that subclasses can use different gate sets.